iothub/device/src/DeviceAuthenticationWithX509Certificate.cs

```diff
@@ -48,6 +48,16 @@ public string DeviceId
         /// </summary>
         public X509Certificate2Collection ChainCertificates { get; }
 
+        /// <summary>
+        /// Dispose necessary objects.
+        /// </summary>
+        public void Dispose()
+        {
+#if !NET451
+            Certificate?.Dispose();
```

Comment:
prefer this approach to avoid disposing twice:

```csharp
Certificate?.Dispose();
Certificate = null;
```

Reply:
Yes thanks for reminding of the best practice. Will do it everywhere.